project-base/src/Controller/Front/ProductController.php

```diff
@@ -259,7 +260,10 @@ public function listByBrandAction(Request $request, $id)
      */
     public function searchAction(Request $request)
     {
-        $searchText = $request->query->get(self::SEARCH_TEXT_PARAMETER, '');
+        $searchText = (string)$request->query->get(self::SEARCH_TEXT_PARAMETER, self::SEARCH_TEXT_DEFAULT_VALUE);
+        if ($this->isRequestSearchTextValid($searchText)) {
```

Comment:
That method actually checks, if the search text is invalid.

Reply:
In invalid state, you should redirect user to homepage and display flash message saying something like: "Search text does not include printable characters".

That is because right now you actually redirect user to page that displays products with space in name and that is weird.

Reply:
> That method actually checks, if the search text is invalid.

> In invalid state, you should redirect user to homepage

both fixed, thanks for good points 👍 

---

> and display flash message saying something like: "Search text does not include printable characters".

not fixed, @sslt agreed with quiet redirect to homepage